pkgs/tools/inputmethods/ibus/fix-paths.patch

```diff
@@ -1,6 +1,41 @@
+diff --git a/bus/services/Makefile.am b/bus/services/Makefile.am
+index 9cc42153..79fdcdbd 100644
+--- a/bus/services/Makefile.am
++++ b/bus/services/Makefile.am
+@@ -43,13 +43,13 @@ systemdservice_in_files = \
+ 
+ if ENABLE_SYSTEMD
+ systemdservice_DATA = $(systemdservice_in_files:.service.in=.service)
+-systemdservicedir = $(SYSTEMD_USER_UNIT_DIR)
++systemdservicedir = $(prefix)$(SYSTEMD_USER_UNIT_DIR)
+ 
+ # in install-data-hook.
+ install-data-hook:
+-	$(MKDIR_P) "$(DESTDIR)$(SYSTEMD_USER_UNIT_DIR)/gnome-session.target.wants/"
++	$(MKDIR_P) "$(systemdservicedir)/gnome-session.target.wants/"
+ 	( \
+-		cd "$(DESTDIR)$(SYSTEMD_USER_UNIT_DIR)/gnome-session.target.wants" ; \
++		cd "$(systemdservicedir)/gnome-session.target.wants" ; \
+ 		$(LN_S) -f "../org.freedesktop.IBus.session.GNOME.service" .; \
+ 	)
+ 	$(NULL)
+diff --git a/configure.ac b/configure.ac
+index a3cdb2da..46666e4f 100644
 --- a/configure.ac
 +++ b/configure.ac
-@@ -429,11 +429,11 @@
+@@ -402,10 +402,7 @@ AC_ARG_ENABLE(systemd-services,
+ AM_CONDITIONAL([ENABLE_SYSTEMD], [test x"$enable_systemd" = x"yes"])
+ 
+ if test x"$enable_systemd" = x"yes"; then
+-    PKG_CHECK_MODULES(SYSTEMD, [
+-        systemd >= 0.7.5
+-    ])
+-    AC_SUBST([SYSTEMD_USER_UNIT_DIR], [`$PKG_CONFIG --variable systemduserunitdir systemd`])
++    AC_SUBST([SYSTEMD_USER_UNIT_DIR], [/lib/systemd/user])
```

Comment:
Either way is fine, though if we keep it in a patch, it should probably go to a separate patch file. `fix-paths.patch` is usually refers to runtime paths in the code, rather than installation paths (yeah, the name is not good).

Reply:
Meh, that sounds like an artificial difference? I think "patching paths due to Nix quirks (or non-Nix quirks)" is a reasonable category

Reply:
In this case, I would consider it an upstream bug, see https://www.bassi.io/articles/2018/03/15/pkg-config-and-paths/. Maybe passing `--define-variable 'prefix=${prefix}'` to the `pkg-config` call would be an upstreamable fix.

Reply:
Oh, I was trying to do something like that earlier, missed `--define-variable`. Yeah that sounds reasonable